sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -82,6 +84,23 @@ singleStatement
     : (statement|setResetStatement) SEMICOLON* EOF
     ;
 
+conditionValue
+    : stringLit
+    | multipartIdentifier
+    ;
+
+conditionValueList
+    : ((conditionValues+=conditionValue (COMMA conditionValues+=conditionValue)*) | SQLEXCEPTION | NOT FOUND)
+    ;
+
+declareCondition
+    : DECLARE multipartIdentifier CONDITION (FOR stringLit)?
+    ;
+
+declareHandler
+    : DECLARE (CONTINUE | EXIT) HANDLER FOR conditionValueList (BEGIN compoundBody END | statement)
```

Comment:
I think we should do `... (BEGIN compoundBody END | statement | setStatementWithOptionalVarKeyword)` to completely match the behavior of `SET` statements with compound bodies.

Reply:
Done.